[docs] Update Netlify deployment docs to include redirect instructions (#19782)
expo/expo: docs/pages/distribution/publishing-websites.mdx

### Manual deployment with the Netlify CDN

> 1. Install the Netlify CLI by running `npm install netlify-cli -g`

Reviewer: ```suggestion
1. Install the Netlify CLI by running `npm install netlify-cli -g`.
```

Add a period to match list styles in this section.

> 2. Build your Expo web app with `npx expo export:web` (SDK 45 and lower should use `expo build:web` via the legacy global Expo CLI).

Reviewer: This step can be removed since it's repeating the set of instructions already covered [Creating a build section](https://docs.expo.dev/distribution/publishing-websites/#creating-a-build).

> 2. Build your Expo web app with `npx expo export:web` (SDK 45 and lower should use `expo build:web` via the legacy global Expo CLI).

Reviewer: This step can be removed since it's repeating the set of instructions already covered [Creating a build section](https://docs.expo.dev/distribution/publishing-websites/#creating-a-build).

> 3. If your app uses routes you must configure `redirects` so the client has full control over the routing logic:

Reviewer: ```suggestion
2. If your app uses routes, add a **_redirects** file inside the **web-build** directory so the client has complete control over the routing logic. This file should contain the `/* /index.html 200` rule.
```

As per our writing style guide, we prefer to use filenames and directory names as bold text: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text

>    3.1. Run `cd web-build`

Reviewer: ```suggestion
   3.1. Run `cd web-build` to navigate inside the directory.
```

>    3.2. Create a `_redirects` file by running `echo "/* /index.html 200" > "\_redirects"`

Reviewer: ```suggestion
   3.2. Run the command `echo "/* /index.html 200" > "_redirects"` to create **_redirects** file with rule.
```

> 4. To deploy:

Reviewer: ```suggestion
3. To deploy:
```

>    4.1. Run `netlify deploy`
> 
>    4.2. Choose `web-build` as the publish directory

Reviewer: These two steps can be as above
```suggestion
   3.1. Run the `netlify deploy` command and choose **web-build** as the publish directory.
```

>    4.3. You should see a URL that you can use to view your project online.

Reviewer: ```suggestion
   3.2. You should see a URL that you can use to view your project online.
```
